### DISCUSSION TITLE
关于QQ官方机器人的适配器的相关问题，如何撤回机器人自己发送的消息

### DISCUSSION THREAD

**Discussion by STESmly** [nonebot/discussions#503] [问答]
找了半天都不知道机器人msg_id怎么获取

**Comment by STESmly** [comment]
测试 = on_command("测试",priority=10, block=True)
@测试.handle()
async def _(event:C2CMessageCreateEvent):
    await 测试.send("测试成功")
    结果 = await Bot.delete_c2c_message(openid=event.get_user_id(), message_id=)  按道理来说应该是这个，撤回私信发的消息

**Comment by yanyongyu** [comment]
消息id在发送请求的返回值里有（[文档](https://bot.q.qq.com/wiki/develop/api-v2/server-inter/message/send-receive/send.html#%E7%BE%A4%E8%81%8A)）
![image](https://github.com/user-attachments/assets/151d39db-c2dd-4b5a-8c40-3b3f96a72047)

所以你可以直接在send的返回值里拿到 `resp = await xxx.send("xxx"); msg_id = resp.id`

**Reply by STESmly** [reply]
但
async def _(event:C2CMessageCreateEvent):
    resp = await 测试.send("测试成功")
    await Bot.delete_c2c_message(openid=event.get_user_id(), message_id=resp.id)
的话会报错
TypeError: __call__() missing 1 required positional argument: 'inst'

**Reply by STESmly** [reply]
id确实能获取到，但样貌和官方文档里面的那个msg_id太不一样了https://bot.q.qq.com/wiki/develop/api-v2/server-inter/message/send-receive/reset.html#%E5%8D%95%E8%81%8A

**Reply by yanyongyu** [reply]
你用Bot类当然不能调用api。。。你先要拿到bot instance。请参考nb文档调用api

**Reply by yanyongyu** [reply]
文档里的只是示例

**Reply by STESmly** [reply]
> 你用Bot类当然不能调用api。。。你先要拿到bot instance。请参考nb文档调用api

哦哦哦懂了懂了，已经运行成功了，谢谢
